simple_install/install.sh.in

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env bash
+
+# ensure we're in the correct folder in case user's CWD is somewhere else
+scriptdir=$(dirname "$0")
+cd "$scriptdir"
+
+# make sure we are not already in a venv
+deactivate
+
+# this should be changed to the tagged release!
+# INVOKE_AI_SRC=https://github.com/invoke-ai/InvokeAI/archive/main.zip
+INVOKE_AI_SRC=https://github.com/invoke-ai/InvokeAI/archive/lstein-simple-install.zip
+INSTRUCTIONS='<insert URL here>'
+TROUBLESHOOTING='<insert URL here>'
+MINIMUM_PYTHON_VERSION=3.9.0
+
+set -euo pipefail
+IFS=$'\n\t'
+
+function _err_exit {
+    if test "$1" -ne 0
+    then
+        echo -e "Error code $1; Error caught was '$2'"
+	if [ "$OS_NAME" == "osx" ]; then
+	    echo "Something went wrong while installing InvokeAI and/or its requirements."
+	    echo "You may need to the Xcode command line tools to proceed. See step number 3 of"
+	    echo "https://invoke-ai.github.io/InvokeAI/INSTALL_SOURCE#walk_through for"
+	    echo "installation instructions and then run this script again."
+	else
+	    echo "Something went wrong while installing InvokeAI and/or its requirements."
+	    echo "See https://invoke-ai.github.io/InvokeAI/INSTALL_SOURCE#troubleshooting for troubleshooting"
+	    echo "tips, or visit https://invoke-ai.github.io/InvokeAI/#installation for alternative"
+	    echo "installation methods"
+	fi
+        read -p "Press any key to exit..."
+        exit
+    fi
+}
+
+
+function version { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+
+echo "InvokeAI simple installer..."
+echo ""
+echo "Some of the installation steps take a long time to run. Please be patient."
+echo "If the script appears to hang for more than 10 minutes, please interrupt with control-C and retry."
+read -n 1 -s -r -p "<Press any key to start the install>"
+echo ""
+
+OS_NAME=$(uname -s)
+case "${OS_NAME}" in
+    Linux*)     OS_NAME="linux";;
+    Darwin*)    OS_NAME="osx";;
+    *)          echo "Unknown OS: $OS_NAME! This script runs only on Linux or Mac" && exit
+esac
+
+OS_ARCH=$(uname -m)
+case "${OS_ARCH}" in
+    x86_64*)    OS_ARCH="64";;
+    arm64*)     OS_ARCH="arm64";;
+    *)          echo "Unknown system architecture: $OS_ARCH! This script runs only on x86_64 or arm64" && exit
+esac
+
+echo "Installing for $OS_NAME-$OS_ARCH"
+# confirm that python is installed and is up to date
+if ! which python &>/dev/null; then
+   echo "Please install Python 3.9 or higher before running this script. See instructions at $INSTRUCTIONS for help."
+   exit -1
+fi
+
+python_version=$(python -V | awk '{ print $2 }')
+if [ $(version $python_version) -lt $(version "$MINIMUM_PYTHON_VERSION") ]; then
+   echo "Installed Python version $python_version is less than the required version $MINIMUM_PYTHON_VERSION. See instructions at $INSTRUCTIONS for help."
+   exit -1
+fi
+
+echo "Python version $python_version detected. Looking good!"
+# install virtualenv support
```

Comment:
Python includes the '[venv](https://docs.python.org/3.10/library/venv.html)' module (since version 3.3). 'virtualenv' is redundant.

Reply:
It is true that `venv` supercedes `virtualenv` , but it's not guaranteed to exist. At least on Ubuntu those are separate `apt install`s:

```
root@63ce946249bd:/# apt update && apt install python3
...
root@63ce946249bd:/# python3 -m venv .venv
The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt install python3.8-venv
    
root@63ce946249bd:/# pip
bash: pip: command not found
root@63ce946249bd:/# python3 -m pip                                                
/usr/bin/python3: No module named pip                                              
root@63ce946249bd:/# apt install python3-venv python3-pip
.... 
:ok: 
```

This is on Ubuntu. But on Fedora, i believe, `python` comes with both `venv` and `pip`. Same on macOS(?) Not sure off the top of my head about Debian or Arch.

This raises the issue of handling different Linux distros with regard to python dependencies... :thinking:

Reply:
The install script calls `pip install virtualenv` as one of the first steps. So this should be ok.

Reply:
There other Ubuntu issue, just to make life difficult, is that it installs python3 as `python3` and not as `python`. One needs to make a symbolic link to get `python` to point at the right place.

I'll add this to the troubleshooting tips.